documents/purchaseorder/bootstrapper.go

```diff
@@ -49,8 +43,18 @@ func (Bootstrapper) Bootstrap(ctx map[string]interface{}) error {
 	}
 	repo.Register(&PurchaseOrder{})
 
+	queueSrv, ok := ctx[bootstrap.BootstrappedQueueServer].(*queue.Server)
+	if !ok {
+		return errors.New("queue server not initialised")
+	}
+
+	txRepo, ok := ctx[transactions.BootstrappedRepo].(transactions.Repository)
```

Comment:
lets just use the txService interface on other packages instead of repo, please. This just breaks that pattern we are creating else where plus there are multiple places in this pr where txService is useful.

Reply:
I think calling it txRepo was a bad idea. but the idea was that we have too many wrappers services which will be behind the other services. Hence, I decided to use it as it is. This will not be breaking any patterns because of following reasons
1. tx repository is behind invoice/po service like the rest of the repositories
2. We still follow transactions handler -> service pattern for rest apis

I would like to move the transaction wait state to transactions packages so that we can re-use the code